File: Viva1/Code Snippets for Error Finding and Missing Code/C++/reverseAnArray.cpp
/*
Problem: Reverse an Array
You are given an array of integers. Complete the function to reverse the array in place.

Constraints:
1 <= n <= 10^5
-10^9 <= arr[i] <= 10^9

Expected Time Complexity: O(n/2)
*/

#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    void reverseArray(vector<int> &arr)
    {
        int left = 0, right = arr.size() - 1;
        while (left < right)
        {
            swap(arr[left], arr[right]);
            left++;
            right--;
        }
    }
};

void printArray(const vector<int> &arr)
{
    for (int x : arr)
    {
        cout << x << " ";
    }
    cout << endl;
}

int main()
{
    vector<int> arr = {1, 2, 3, 4, 5};
    Solution sol;

    cout << "Original array: ";
    printArray(arr);

    sol.reverseArray(arr);

    cout << "Reversed array: ";
    printArray(arr);

    return 0;
}

File: Viva1/Code Snippets for Error Finding and Missing Code/Python/reverse_an_array.py
'''
Problem: Reverse an Array
You are given an array of integers. Complete the function to reverse the array in place.

Constraints:
1 <= n <= 10^5
-10^9 <= arr[i] <= 10^9

Expected Time Complexity: O(n/2)
'''

class Solution:
    def reverseArray(self, arr):
        left, right = 0, len(arr) - 1
        while left < right:
            # Swap elements at left and right indices
            arr[left], arr[right] = arr[right], arr[left]
            left += 1
            right -= 1

def printArray(arr):
    print(" ".join(map(str, arr)))

if __name__ == "__main__":
    arr = [1, 2, 3, 4, 5]
    sol = Solution()

    print("Original array: ", end="")
    printArray(arr)

    sol.reverseArray(arr)

    print("Reversed array: ", end="")
    printArray(arr)